homeassistant/components/jewish_calendar/__init__.py

```diff
@@ -157,3 +170,26 @@ async def async_unload_entry(hass: HomeAssistant, config_entry: ConfigEntry) ->
         hass.data[DOMAIN].pop(config_entry.entry_id)
 
     return unload_ok
+
+
+@callback
+def async_update_unique_id(
```

Comment:
I think you can use `async_migrate_entries()` https://github.com/home-assistant/core/blob/dev/homeassistant/helpers/entity_registry.py#L1567

It will resolve all entities from the entity registry from the config entry and apply updates from a callback.

Reply:
Or in minimum use the `async_entries_for_config_entry` helper to get the entities instead of listing from the platform.

Reply:
I discussed that with @joostlek on Discord. During `async_setup_entry` at this point, no one has a config entry as this was only added a few days ago. So calling `async_entries_for_config_entry` will result in an empty list. Unless the user reloaded the integration after the initial import.
I tried more or less the implementation he used in the WAQI integration and it failed on the test.
That's also the reason not to use `async_migrate_entries` as users still don't have config entries for the Jewish Calendar integration.

Reply:
Of course they have a config entry as `async_setup_entry` is only loaded from config entries.
So if they don't have a config entry they would anyhow not reach this code

Reply:
Right, but `async_entries_for_config_entry` returns empty on the initial try since the entity registry was not updated yet.

Reply:
LOL. You're of course right. 👍 

Anyhow get the entities from the entity registry for this integration instead of loading from `PLATFORM_DESCRIPTIONS`. That was actually what I wanted from the beginning before I started mixing in the other functions.

Reply:
The reason for the `PLATFORM_DESCRIPTIONS` is because I need the description key to build the existing unique_id when I'm trying to get the existing entities.
Is there a way to get all the entities without providing the existing unique_ids from the entity registry? Like just by providing the platform and the domain?
If so I can loop in the method over all of them without needing the descriptions and simply replacing the prefix.

Reply:
AFAIK, I need to provide the unique id

Reply:
I think I'm overcomplicating this so let's stop here. I'll do a review from scratch.